src/main/java/org/example/Main.java

```diff
@@ -121,339 +177,256 @@ public boolean isHappy() {
-                        if (users.contains("-SUNNY") && num.sunny) {
-                            continue;
-                        }
-                        if (users.contains("-SQUARE") && num.square) {
-                            continue;
-                        }
-                        if (users.contains("-JUMPING") && num.jumping) {
-                            continue;
-                        }
-                        if (users.contains("-HAPPY") && num.happy) {
-                            continue;
-                        }
-                        if (users.contains("-SAD") && num.sad) {
-                            continue;
-                        }
-                        num.printShortInfo();
-                        count--;
-                    }
-                }
+        Set<String> allProperties =
+                Stream.of(MyNumber.Params.values()).map(MyNumber.Params::name).collect(Collectors.toSet());
+        Set<String> allNegativeProperties =
+                Stream.of(MyNumber.Params.values()).map(MyNumber.Params::name).map(x -> "-" + x).collect(Collectors.toSet());
+        Collection<String> incorrectProperties = new ArrayList<>();
+        for (String property : properties) {
+            if (!(allProperties.contains(property) || allNegativeProperties.contains(property))) {
+                incorrectProperties.add(property);
             }
         }
+
+        if (incorrectProperties.isEmpty()) {
+            return false;
+        }
+        if (incorrectProperties.size() == 1) {
+            System.out.println(MessageFormat.format("The property [{0}] is wrong.", incorrectProperties));
+        } else {
+            System.out.println(MessageFormat.format("The properties [{0}] are wrong.",
+                    String.join(", ", incorrectProperties)));
+        }
+
+        System.out.println("Available properties: " + Arrays.stream(MyNumber.Params.values()).toList());
+        return true;
+    }
+
+    private static boolean hasExclusiveProperty(Collection<String> properties) {
+        if (properties.isEmpty()) {
+            return false;
+        }
+
+        boolean hasExclProperty = false;
+        for (var exclPropertyPair : MyNumber.getExclusiveProperties()) {
+            if (properties.contains(exclPropertyPair[0]) && properties.contains(exclPropertyPair[1])) {
+                System.out.println(MessageFormat.format("The request contains mutually exclusive properties: [{0}, {1}].",
+                        exclPropertyPair[0], exclPropertyPair[1]));
+                hasExclProperty = true;
+            }
+        }
+        for (String property : properties) {
+            if (properties.contains("-" + property)) {
+                System.out.println(MessageFormat.format("The request contains mutually exclusive properties: [{0}, {1}].",
+                        property, "-" + property));
+                hasExclProperty = true;
+            }
+        }
+        if (hasExclProperty) {
```

Comment:
I think this if is incorrect (or the message inside)

Reply:
In task description we need to print the error message and warn the user:
```
Enter a request: 1 5 sunny square

The request contains mutually exclusive properties: [SQUARE, SUNNY]
There are no numbers with these properties.
```
But they do not check it in tests.